doc/sphinx/source/community/detailed_release_procedure.rst

```diff
@@ -0,0 +1,190 @@
+
+Below is a list of steps that the release manager, together with the previous release manager, should go through before the actual release;
+these include testing the new code by running all available recipes in yhe `main` branch, and comparing the output against
+the previous release.
+
+Open an issue on GitHub
+-----------------------
+
+First, open an issue on GitHub where the testing workflow before the release is documented (see example https://github.com/ESMValGroup/ESMValTool/issues/2881).
+Name it something relevant like "Recipe testing and comparison for release 2.x.x", and populate the issue description with information
+about where the testing is taking place, what tools are used, and what versions, here are some suggestions:
+
+
+Where the work is done e.g. work done on DKRZ/Levante:
+
+- submit files in `/home/b/b382109/submit` (for v2.7.0)
+- output in `/scratch/b/b382109/esmvaltool_output` (for v2.7.0)
+
+We should document various utilities' versions so that the work can be reproduced in case there
+is an issue, or release work needs to be picked up mid-release by another release manager:
+
+- documenting `conda`/`mamba` versions:
+
+.. code-block:: bash
+
+  mamba --version
+
+- documenting `git` branch and its state:
+
+.. code-block:: bash
+
+  git status
+
+Furthermore, the runtime environment needs to be documented:
+
+.. code-block:: bash
+
+  mamba env create -n tool2xxTest -f environment.yml
+  conda activate tool2xxTest
+
+Make a copy of the environment file, and attach it in the release testing issue; to
+record the environment in a yaml file use e.g. `conda env export > ToolEnv270Test.txt`.
+
+Mention any special data movement one needs to perfom to run the tests e.g.:
+for v2.7.0 the release manager moved the autoassess-specific files to
+`/home/b/$USER/autoassess_files` on DKRZ/Levante.
+
+Modifications to configuration files need to be documented as well: for example,
+for v2.7.0 the release manager had to modify `config-user.yml` file by
+adding a number of extra paths for DKRZ/Levante-specific data pools:
+
+.. code-block:: yaml
+
+  CMIP6:
+    - /work/bd0854/DATA/ESMValTool2/CMIP6_DKRZ
+    - /work/bd0854/DATA/ESMValTool2/download/CMIP6
+  CMIP5:
+    - /work/bd0854/DATA/ESMValTool2/CMIP5_DKRZ
+    - /work/bd0854/b309141/additional_CMIP5
+    - /work/bd0854/DATA/ESMValTool2/download/cmip5/output1
+    - /work/bd0854/DATA/ESMValTool2/download/cmip5
+
+Submit run scripts - test recipe runs
+-------------------------------------
+
+We are now ready to start running all the available recipes, to compare output against previous release. Running is currently done
+via batch scripts submitted to a schedulers (SLURM). Generate the submission scripts using the `generate.py` Python script;
+you can find a copy of the script either in `/home/b/b382109/Tool_Release_270_Scripts` or
+in the draft Pull Request https://github.com/ESMValGroup/ESMValTool/pull/2883.
```

Comment:
This is something that needs to be discussed with the @ESMValGroup/technical-lead-development-team. Do we start our own set of scripts or can we use an established job submission system?

Reply:
this uses SLURM (as job submission system/scheduler) - the problem is how we set up the jobs - either with these sort of scripts or with automated test scripts (formerly started as development by the MO, unceremoniously postponed until further notice)

Reply:
Yes, but there is existing software that can write such scripts (e.g. cylc, [xenon](https://github.com/xenon-middleware/pyxenon), [Dirac](https://dirac.readthedocs.io/en/latest/UserGuide/GettingStarted/UserJobs/DiracAPI/index.html), and probably a lot more). Writing your own scripts is reinventing the wheel and should only be done for very simple things.